geode-core/src/main/java/org/apache/geode/cache/query/security/GeodeBasedMethodAuthorizer.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.security;
+
+import java.lang.reflect.Method;
+import java.util.Objects;
+import java.util.Properties;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Declarable;
+import org.apache.geode.cache.Region;
+
+/**
+ * An immutable and thread-safe {@link MethodInvocationAuthorizer} that allows any method execution
+ * as long as the target object does not belong to a Geode package, or does belong but it's marked
+ * as
+ * safe (see {@link RestrictedMethodAuthorizer#isAllowedGeodeMethod(Method, Object)}).
+ * <p/>
+ *
+ * Some known dangerous methods (like {@link Object#getClass()}) are also rejected by this
+ * {@link MethodInvocationAuthorizer} implementation, no matter whether the target object belongs
```

Comment:
I see this pattern repeated many times: "this {@link MethodInvocationAuthorizer} implementation". I would change all of these to just "this implementation". Your first javadoc sentence linked to MethodInvocationAuthorizer and of course this class implements that interface so I think that is enough linking to MethodInvocationAuthorizer